src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -150,7 +153,39 @@ private void OnGetSuggestionsCommandExecute()
                 _suggestionsDropdownTelemetry.DropdownButtonClickTimes += 1;
             }
 
-            _getSuggestionsTask = _smartRenameSession.GetSuggestionsAsync(_cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
+            var allRenameLocations = BaseViewModel.Session.AllRenameLocationsTask.Join();
+            ImmutableDictionary<string, string[]> context = default;
+            if (allRenameLocations.Locations.Count > 0)
+            {
+                var references = new List<string>();
+                foreach (var renameLocation in allRenameLocations.Locations)
+                {
+                    var syntaxTree = renameLocation.Document.GetSyntaxTreeSynchronously(_cancellationTokenSource.Token);
```

Comment:
this is a blocker.   The above assert says this is on the UI thread.  But getting syntax trees is *expensive*.  So thsi will block the UI thread.  We need to make this async/non-blocking.

Reply:
I agree. I'm still at the proof of concept stage

Reply:
Ok.  Want to move to draft in the meantime? :)

Reply:
Sure!